packages/shared/invokeGuardedCallback.js

```diff
@@ -173,6 +179,12 @@ if (__DEV__) {
         this._caughtError = null;
       }
 
+      if (shouldIgnoreError && error != null) {
+        this._hasSuppressedError = true;
+        this._suppressedError = error;
+        error.suppressReactErrorLogging = true;
```

Comment:
We should remove this concept completely.

Reply:
This will require changing our unit test setup [here](https://github.com/facebook/react/blob/48833f698dc28a9af09819ba731a2e94d5bf9da7/scripts/jest/setupEnvironment.js#L43-L45) and [here](https://github.com/facebook/react/blob/48833f698dc28a9af09819ba731a2e94d5bf9da7/scripts/jest/setupEnvironment.js#L33-L37) not to rely on this property.

Instead we can try changing the `expect().toThrow()` Jest matcher to set up an `'error'` handler that does `preventDefault()`. Or figure out some other way to avoid failing all tests that intentionally throw an error.

Reply:
I added a functionality but I am not able to figure out the `resets instance variables before unmounting failed node` test is failing saying `Expected test not to call console.error().`
am i missing something?